src/mlpack/methods/hmm/hmm_model.hpp

```diff
@@ -39,14 +41,17 @@ class HMMModel
   HMM<distribution::GaussianDistribution>* gaussianHMM;
   //! Not used if type is not GaussianMixtureModelHMM.
   HMM<gmm::GMM>* gmmHMM;
+  //! Not used if type is not DiagonalGaussianMixtureModelHMM.
+  HMM<gmm::DiagonalGMM>* diagGMMHMM;
 
  public:
   //! Construct an uninitialized model.
   HMMModel() :
```

Comment:
What about we use a default value for the second constructor and remove this one?